### PR TITLE
Support custom scroll behaviour

### PR DESCRIPTION
**Pull request by zamotany**
Ref: #1012

### CODE REVIEW

File: packages/gluestick/shared/containers/Root.js [hunk 1]
```diff
@@ -73,28 +73,39 @@ export default class Root extends Component<DefaultProps, Props, State> {
...
+        // behaviour.
+        const { useScroll: customScrollBehavior } = routes.find(route => (
+          route.useScroll &&
+          prevRouterProps &&
+          prevRouterProps.location.pathname !== location.pathname
+        )) || {};
+
+        if (typeof customScrollBehavior === 'function') {
+          return customScrollBehavior(prevRouterProps, location);
+        }
```
Comment (ferrannp) [inline packages/gluestick/shared/containers/Root.js]: This should be a `function` as part of the API. To ensure this use Flow and/or propTypes. We do not need such checkings in the code (neither tests).

**Review by ferrannp — CHANGES_REQUESTED** [review]

File: packages/gluestick/shared/containers/Root.js (hunk 1 above)
Comment (zamotany) [reply packages/gluestick/shared/containers/Root.js]: - Can't use `propTypes`, it's Root component but `useScroll` can be defined on Route component.
- Flow types won't cover it - I've checked - I even changed build process to output `Root.js.flow` file to project so it'll get properly check but still flow doesn't detect error.

Comment (ferrannp) [reply packages/gluestick/shared/containers/Root.js]: Still, our API it says it should be a function right? I prefer the code to crash if user passes let's say a boolean that just silently ignore it.

Comment (zamotany) [reply packages/gluestick/shared/containers/Root.js]: Have you seen how helpless are stack traces for some errors? It's better to have that check and add `else` block with meaningful error message - so just by reading it, you would know what you have to fix.

**Pull request merged**